source-academy/cadet-frontend
# Mobile Progressive Web App

### Description


This PR improves Progressive Web App functionality of Source Academy:
- Added iOS and Android icons and consolidated all icons under the '/public/icons' directory
- Added 'portrait-primary' screen lock to prevent user from turning to landscape mode (we are disabling the use of the app in landscape mode) --> Note that this screen orientation lock is not supported in Safari browsers
- Updated default app name from 'Cadet' to 'Source'
- Updated start URL of app manifest.json to match that of the app

Android (icons may differ due to Android adaptive icons)
<img width="107" alt="image" src="https://user-images.githubusercontent.com/53928333/107171667-50044980-69fe-11eb-8170-e2c214bfbf62.png">

iOS
<img width="121" alt="image" src="https://user-images.githubusercontent.com/53928333/107171722-690cfa80-69fe-11eb-9a31-cced1aacc174.png">

### Type of change


### How to test



Add/ Install the app to homescreen. The defaults should reflect the above screenshots (i.e. app name is 'Source', and icons should show). Install instructions can be found [here](https://github.com/source-academy/cadet-frontend/wiki/Mobile-Development-Guide#setting-up-the-progressive-web-app-on-your-mobile-device).

Please pull the PR and use the localhost version. Alternatively, the current HEAD of this PR is deployed [here](https://sa-mobile-git-mobile-pwa.chownces.vercel.app/)

### Checklist
